### PR TITLE
Allow Justice, Wrath and Whiplash to target enemies MC'd by XCOM

### LINKED ISSUE #1291: Some Skirmisher abilities can not target enemies Mind Controlled by XCOM
Skirmisher's Justice, Wrath and Whiplash can not target enemies that have been Mind Controlled by XCOM. There is probably no particular reason why not, it may be just an oversight from Firaxis.

To reproduce:
1. Start a tactical mission with a Skirmisher possessing one of the offending skills
1. `GiveReward Deception_T2` until you Mind Control an enemy that would be an eligible target (if not for the issue)
1. `ttc` the Skirmisher into range of that unit
1. Select the offending skill from hotbar
1. Observe whether the Mind Controlled unit shows up as a valid target or not.

Found in LWOTC, but examining base game code I'm fairly confident the issue is also there, hence the CHL issue.

### PR DESCRIPTION
**Pull request by copyrite**
Fixes #1291

### CODE REVIEW

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/X2Ability_SkirmisherAbilitySet.uc [hunk 1]
```diff
@@ -823,6 +845,11 @@ static function X2AbilityTemplate Whiplash()
 	UnitPropertyCondition = new class'X2Condition_UnitProperty';
 	UnitPropertyCondition.ExcludeRobotic = false;
 	UnitPropertyCondition.ExcludeOrganic = true;
+
+	/// HL-Docs: ref:Bugfixes; issue:1293
+	/// Allow Whiplash to target enemies Mind Controlled by XCOM
+	UnitPropertyCondition.TreatMindControlledSquadmateAsHostile = true;
```
Comment (Iridar) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/X2Ability_SkirmisherAbilitySet.uc:851]: I'm reasonably sure `TreatMindControlledSquadmateAsHostile` is responsible for allowing the player to target XCOM squadmates that are mindcontrolled by the enemy, not for allowing to target enemy units mindcontrolled by XCOM. Did you test this fix?

Comment (copyrite) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/X2Ability_SkirmisherAbilitySet.uc:851]: I did test this, and tested again after this review. First, the good:

Humanoid ADVENT are targetable with Justice with the fix:


And they take damage after the pull:


However, you may be right about `TreatMindControlledSquadmateAsHostile` still, it doesn't show up on known working examples. `LivingHostileTargetProperty` could be a candidate, it's suspiciously absent from `SkirmisherAbilitySet`. Will get back to you.

Comment (copyrite) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/X2Ability_SkirmisherAbilitySet.uc:851]: Zelfana backs this implementation [on Discord](https://discord.com/channels/165245941664710656/167912351763398657/1176972883659079771):
> > var() bool    TreatMindControlledSquadmateAsHostile;   //  if a squadmate is mind controlled (i.e. it used to be your enemy), normally they are considered friendly. consider them hostile instead.
> 
> this is exactly what lets you attack mind-controlled enemies

> a mind-controlled soldier is just treated as hostile by default

> in the skirmisher skills it should be toggled on the existing unit conditions, which you managed to do and have it let you target mind-controlled enemies

Comment (copyrite) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/X2Ability_SkirmisherAbilitySet.uc:845]: Found an improvement, went ahead and squashed it.

**Pull request merged**